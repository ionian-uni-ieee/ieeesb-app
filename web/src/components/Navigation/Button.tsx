import React from 'react'
import { Link } from 'react-router-dom'
import { scrollToID } from 'utils/browserHelpers'

interface IButtonProps {
  label: string
  href?: string
  onClick?(e: React.MouseEvent<HTMLAnchorElement>): void
}

const Button: React.FC<IButtonProps> = props => {
  const {
    label,
    href,
    onClick= null,
  } = props

  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>): void => {
    if (onClick) {
      onClick(e)
    }
    if (!href) {
      return
    }
    e.preventDefault()
    scrollToID(href)
  }

  return (
    <Link
      to={href || 'ieee'}
      className='ui navigation__button'
      onClick={handleClick}
    >
      {label}
    </Link>
  )
}

export default Button
