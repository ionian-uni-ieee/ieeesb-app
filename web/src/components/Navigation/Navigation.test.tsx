import React from 'react'
import { MemoryRouter } from 'react-router'

import { render } from '@testing-library/react'

import Navigation from './Navigation'

describe('Navigation', () => {
  it('should render', () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation>
          <Navigation.Button href='#' label='hi' />
        </Navigation>
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
