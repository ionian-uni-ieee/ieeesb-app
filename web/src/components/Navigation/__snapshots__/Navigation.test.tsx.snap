// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render 1`] = `
<div>
  <div
    class="ui navigation"
  >
    <a
      class="ui navigation__button"
      href="/"
    >
      hi
    </a>
  </div>
</div>
`;
